beets/dbcore/db.py

```diff
@@ -680,14 +684,17 @@ def mutate(self, statement, subvals=()):
         """Execute an SQL statement with substitution values and return
         the row ID of the last affected row.
         """
-        cursor = self.db._connection().execute(statement, subvals)
-        return cursor.lastrowid
+        try:
+            cursor = self.db._connection().execute(statement, subvals)
+            return cursor.lastrowid
+        except sqlite3.OperationalError as e:
+            raise AccessFileError("unable to open database file. It might be a permissions problem")
```

Comment:
Yep! Or we can just have one `if` that checks whether it's either of those things:

```
if e.args[0] in ('string1', 'string2'):
```

I don't quite understand what you mean about the UserError… is some *other* code producing a UserError exception?